api_tests/nodes/views/test_node_list.py

```diff
@@ -346,6 +376,31 @@ def test_incorrect_filtering_field_not_logged_in(self):
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], "'notafield' is not a valid field for this endpoint.")
 
+    def test_filtering_on_root(self):
+        # Build up a family of nodes
+        node_structure = [5, 2, [1, 2], 3, [10, 2], 2, [4, [2, 2], 6]]
+        render_generations_from_node_structure_list(self.project_one, node_structure)
```

Comment:
Is it necessary to create such a complex node structure? Not only will creating 42 nodes slow down the test, but it will also make it harder to debug the test if it fails.